packages/blockchain-wallet-v4-frontend/src/modals/Swap/PreviewSwap/index.tsx

```diff
@@ -1,345 +1,74 @@
-import React, { PureComponent } from 'react'
-import { FormattedMessage } from 'react-intl'
-import { connect, ConnectedProps } from 'react-redux'
-import { bindActionCreators, compose, Dispatch } from 'redux'
-import { Form, InjectedFormProps, reduxForm } from 'redux-form'
-import styled from 'styled-components'
+import React, { useEffect } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { compose } from 'redux'
+import { InjectedFormProps, reduxForm } from 'redux-form'
 
-import { formatCoin } from '@core/exchange/utils'
-import {
-  Button,
-  HeartbeatLoader,
-  Icon,
-  Link,
-  SkeletonRectangle,
-  Text,
-  TextGroup
-} from 'blockchain-info-components'
-import { ErrorCartridge } from 'components/Cartridge'
-import { FlyoutWrapper, Row, Title, Value } from 'components/Flyout'
+import { FlyoutOopsError } from 'components/Flyout/Errors'
 import { GenericNabuErrorFlyout } from 'components/GenericNabuErrorFlyout'
-import { actions, selectors } from 'data'
-import { RootState } from 'data/rootReducer'
-import { Analytics, InitSwapFormValuesType, SwapAmountFormValues } from 'data/types'
+import { actions } from 'data'
+import { Analytics } from 'data/types'
 import { isNabuError } from 'services/errors'
 
 import { Props as BaseProps, SuccessStateType } from '..'
-import { FeeBreakdownBox, FromToLogoLeft, TopText } from '../components'
+import Loading from '../template.loading'
+import { PreviewSwapSuccess } from './PreviewSwapSuccess'
+import { getData } from './selectors'
 
-const StyledRow = styled(Row)`
-  display: flex;
-  flex-direction: column;
-  justify-content: flex-start;
-  align-items: center;
-  padding-top: 30px;
-  padding-bottom: 24px;
+const PreviewSwap = (props: Props) => {
+  const dispatch = useDispatch()
```

Comment:
This would be a great chance to refactor out all the complexity and interaction with Redux to a hook. You can refer to `ConfirmingBuyOrder.tsx` as an example.